lucio-desimone-globant/golang-bootcamp
# [BGO-4] Solving comments and making enhancements

# Description

* Changes in the implementation:
    * use UUID as id
    * correct use of interfaces
    * beer package added
    * testing approach
  
* Comments solved:
    * Beer should have its own package.
    * Make the DB operations as `operate(c_mongo, beers)`
    * Don't use interface{} because it has a complex implementation for this course. Use beer instead
    * Use an interface for DB operations
    * Use a function to create beers
    * Use error instead of booleans in queries
    * Use test functions for each test case for example func TestCreateOK(t *testing.T)

# Notes

* A test case is missing. I expect to be learning how to test third-party libraries soon

# Next steps 

* Continue with the next exercise using the file system instead of the heap
